apps/extension/src/components/password-input.tsx

```diff
@@ -0,0 +1,75 @@
+import { EyeClosedIcon, EyeOpenIcon } from '@radix-ui/react-icons';
+import { useMemo, useState } from 'react';
+import { Input, InputProps } from 'ui/components';
+import { cn } from 'ui/lib/utils';
+
+interface Validation {
+  checkFn: (txt: string) => boolean;
+  type: 'warn' | 'error'; // corresponds to red or yellow
+  error: string;
+}
+
+interface PasswordInputProps {
+  passwordValue: string;
+  label: string;
+  validations: Validation[];
+  onChange: InputProps['onChange'];
+}
+
+export const PasswordInput = ({
+  passwordValue,
+  label,
+  validations,
+  onChange,
+}: PasswordInputProps) => {
+  const [reveal, setReveal] = useState(false);
+
+  const validationResult = useMemo(() => {
+    const validation = validations.map(i => ({
+      check: i.checkFn(passwordValue),
+      variant: i.type,
+      error: i.error,
+    }));
+
+    const error = validation.find(i => i.variant === 'error' && i.check);
+    if (error) return error;
+
+    const warn = validation.find(i => i.variant === 'warn' && i.check);
+    if (warn) return warn;
+
+    return null;
+  }, [validations, passwordValue]);
```

Comment:
Think this could be simpler:

```typescript
const priorityResult = useMemo(() => {
    const results = validations.filter(v => v.checkFn(passwordValue));
    const error = results.find(v => v.type === 'error');
    return error ? error : results.find(v => v.type === 'warn');
}, [validations, passwordValue]);
```